darwin/future/meta/objects/workflow.py

```diff
@@ -55,6 +55,12 @@ def stages(self) -> StageQuery:
             meta_params["dataset_name"] = self.datasets[0].name
         return StageQuery(self.client, meta_params=meta_params)
 
+    def _get_dataset_stage(self) -> Stage | None:
```

Comment:
I'm torn between an optional None return here or raising some MissingDataset Exception or similar. I think most functions that would call this would expect it to return something useful and if not then it should abort process and be captured somewhere.

Reply:
Yep that makes sense. Let me raise `MissingDataset` Exception when there's no Dataset Stage in a workflow (this will replace the returning `None` from this method)